### PR TITLE
Determine data type of target tensor during runtime.

### PR DESCRIPTION
**Pull request by mmanzoorTT**

### CODE REVIEW

File: runtime/lib/ttnn/runtime.cpp [hunk 1]
```diff
@@ -50,6 +50,12 @@ Tensor createTensor(std::shared_ptr<void> data,
   return Tensor(tensor, data, DeviceRuntime::TTNN);
 }
 
+tt::target::DataType getTensorDataType(Tensor tensor) {
+  std::shared_ptr<::ttnn::Tensor> nnTensor =
+      std::static_pointer_cast<::ttnn::Tensor>(tensor.handle);
```
Comment (jnie-TT) [inline runtime/lib/ttnn/runtime.cpp:55]: Can we do `tensor.as<::ttnn::Tensor>` here?

Comment (mmanzoorTT) [reply runtime/lib/ttnn/runtime.cpp]: Thanks for the suggestion. updated.

File: runtime/lib/ttnn/runtime.cpp [hunk 2]
```diff
@@ -50,6 +50,11 @@ Tensor createTensor(std::shared_ptr<void> data,
   return Tensor(tensor, data, DeviceRuntime::TTNN);
 }
 
+tt::target::DataType getTensorDataType(Tensor tensor) {
+  ::ttnn::Tensor nnTensor = tensor.as<::ttnn::Tensor>(DeviceRuntime::TTNN);
```
Comment (jnie-TT) [inline runtime/lib/ttnn/runtime.cpp:54]: Thanks for the update. Can we make this a const reference just so we don't copy the ttnn tensor, which can be costly?

Comment (mmanzoorTT) [reply runtime/lib/ttnn/runtime.cpp]: updated as suggested.

**Review by jnie-TT — APPROVED** [review]
Thanks Asif! TTNN runtime changes lgtm

**Pull request merged**